src/game/level/level_editor.c

```diff
@@ -388,12 +389,15 @@ int level_editor_saveas_event(LevelEditor *level_editor,
     case SDL_KEYDOWN: {
         if (event->key.keysym.sym == SDLK_RETURN) {
             trace_assert(level_editor->file_name == NULL);
+            char path[LEVEL_FOLDER_MAX_LENGTH];
+            snprintf(
+                path,
+                LEVEL_FOLDER_MAX_LENGTH,
+                "./assets/levels//%s.txt",
+                edit_field_as_text(level_editor->edit_field_filename));
             level_editor->file_name = PUSH_LT(
                 level_editor->lt,
-                string_duplicate(
-                    edit_field_as_text(
-                        level_editor->edit_field_filename),
-                    NULL),
+                path,
```

Comment:
Are you seriously pushing a string allocated on the stack to a Lifetime object? That's a pretty reliable way to corrupt your stack. Please, at least duplicate it or something.